sys/net/network_layer/ng_ipv6/ext/ng_ipv6_ext.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <errno.h>
+
+#include "utlist.h"
+#include "net/ng_pktbuf.h"
+#include "net/ng_ipv6.h"
+
+#include "net/ng_ipv6/ext.h"
+
+bool ng_ipv6_ext_demux(kernel_pid_t iface, ng_pktsnip_t *pkt,
+                       uint8_t nh)
+{
+    ng_pktsnip_t *ext_snip;
+    ng_ipv6_ext_t *ext;
+    unsigned int offset = 0;
+
+    ext = ((ng_ipv6_ext_t *)(((uint8_t *)pkt->data) + sizeof(ng_ipv6_hdr_t)));
+
+    bool c = true;
+
+    while (c) {
+        switch (nh) {
+            case NG_PROTNUM_IPV6_EXT_HOPOPT:
+            case NG_PROTNUM_IPV6_EXT_DST:
+            case NG_PROTNUM_IPV6_EXT_RH:
+            case NG_PROTNUM_IPV6_EXT_FRAG:
+            case NG_PROTNUM_IPV6_EXT_AH:
+            case NG_PROTNUM_IPV6_EXT_ESP:
+            case NG_PROTNUM_IPV6_EXT_MOB:
+                /* TODO: add handling of types */
+                nh = ext->nh;
+                offset += ((ext->len * NG_IPV6_EXT_LEN_UNIT) + NG_IPV6_EXT_LEN_UNIT);
+                ext = ng_ipv6_ext_get_next((ng_ipv6_ext_t *)ext);
+                break;
+
+            default:
+                c = false;
```

Comment:
Or are you complaining about the fact that I don't really do something with the extension headers (the the position of your line comment is odd). I'm not really sure that this is what is meant by "unrecognized". As soon as I encounter one of the already defined extension headers by IANA, I know what to do, regardless if I interpret it or not. So I don't need to error on this (it's a kind of forward compatibility). If however a new extension header (or any next header) which length is introduced that is unknown to this implementation the packet is discarded (ICMPv6 error messages are not send in any case at the moment)